mailbox/tools/indexer/src/main/java/org/apache/mailbox/tools/indexer/ReIndexerImpl.java

```diff
@@ -137,10 +136,10 @@ private void handleMailboxIndexingIterations(MailboxSession mailboxSession, Mail
         while (iterator.hasNext()) {
             MailboxMessage message = iterator.next();
             ImpactingMessageEvent impactingMessageEvent = findMostRelevant(mailboxRegistration.getImpactingEvents(message.getUid()));
-            if (impactingMessageEvent == null) {
```

Comment:
you don't check for impactingMessageEvent == null anymore ?

Reply:
Latter on this is replaced by an Optional...

I may have messed up an intermediary refactoring step.